lib/mongodb.js

```diff
@@ -1055,12 +1055,26 @@ MongoDB.prototype.count = function count(model, where, options, callback) {
  */
 MongoDB.prototype.replaceById = function replace(model, id, data, options, cb) {
   if (this.debug) debug('replace', model, id, data);
+  var self = this;
   var oid = this.coerceId(model, id);
-  this.replaceWithOptions(model, oid, data, { upsert: false }, function(err, data) {
-    cb(err, data);
+  this.find(model, oid, options, function(err, foundInstance) {
```

Comment:
Same comment as in https://github.com/strongloop/loopback-connector/pull/73#pullrequestreview-16943741

You need to detect the outcome of the underlying replace command to decide whether an existing record was updated or whether there was no matching record found.